packages/bindings/src/test_utils.rs

```diff
@@ -0,0 +1,1146 @@
+//! Contains integration test utils for desmos custom modules.
+
+use crate::msg::DesmosMsg;
+use crate::query::DesmosQuery;
+use anyhow::Result as AnyResult;
+use cosmwasm_std::testing::{MockApi, MockStorage};
+use cosmwasm_std::{Addr, Api, Binary, BlockInfo, Empty, Event, Querier, Storage};
+use cw_multi_test::{
+    App, AppResponse, BankKeeper, BasicApp, BasicAppBuilder, CosmosRouter, FailingDistribution,
+    FailingStaking, Module, Router, WasmKeeper,
+};
+use std::convert::TryFrom;
+
+#[cfg(feature = "posts")]
+use crate::posts::{mocks::mock_posts_query_response, msg::PostsMsg};
+#[cfg(feature = "profiles")]
+use crate::profiles::{mocks::mock_profiles_query_response, msg::ProfilesMsg};
+#[cfg(feature = "reactions")]
+use crate::reactions::{mocks::mock_reactions_query_response, msg::ReactionsMsg};
+#[cfg(feature = "relationships")]
+use crate::relationships::{mocks::mock_relationships_query_response, msg::RelationshipsMsg};
+#[cfg(feature = "reports")]
+use crate::reports::{mocks::mock_reports_query_response, models::ReportTarget, msg::ReportsMsg};
+#[cfg(feature = "subspaces")]
+use crate::subspaces::{mocks::mock_subspaces_query_response, msg::SubspacesMsg};
+
+/// DesmosApp wraps the desmos custom module into a mock app for integration tests.
+/// It always returns successful response with proper events.
+pub type DesmosApp =
+    App<BankKeeper, MockApi, MockStorage, DesmosKeeper, WasmKeeper<DesmosMsg, DesmosQuery>>;
+
+/// Represents the implementation of [`Module`](cw_multi_test::Module) for handling the desmos execution and query messages.
+#[derive(Default)]
+pub struct DesmosKeeper {}
```

Comment:
Instead of implementing the whole state on desmos chain, I make `DesmosKeeper` as a module which always returns the successful response with proper events. Devs can handle event in the `reply` phase as the [sample](https://docs.cosmwasm.com/docs/1.0/smart-contracts/message/submessage/#handling-a-reply-from-a-token-transfer).